src/Lykke.Service.PayInvoicePortal.Services/PaymentsService.cs

```diff
@@ -29,6 +29,17 @@ ILykkeAssetsResolver lykkeAssetsResolver
             _lykkeAssetsResolver = lykkeAssetsResolver;
         }
 
+        public async Task<Payment> GetByInvoiceId(string invoiceId)
+        {
+            InvoiceModel invoice = await _payInvoiceClient.GetInvoiceAsync(invoiceId);
+
+            var payment = Mapper.Map<Payment>(invoice);
+
+            payment.SettlementAsset = await _lykkeAssetsResolver.TryGetAssetAsync(payment.SettlementAssetId);
```

Comment:
Is NRE possible here?

Reply:
Is it expected that SettlementAsset can be null here?

Reply:
NRE not possible, can be null only because of misconfiguration after previous right configuration which possibility is very low.